backend/mock/propelauth_fastapi_mock.py

```
@@ -0,0 +1,596 @@
+        auth_url: str,
+        integration_api_key: str,
+        token_verification_metadata: TokenVerificationMetadata | None,
+        debug_mode: bool,
+    ):
+        self.auth_url = auth_url
+        self.integration_api_key = integration_api_key
+        self.token_verification_metadata = token_verification_metadata
+        self.debug_mode = debug_mode
+        self.auth = Auth(auth_url, integration_api_key, None)
+
+    def require_user(self, credentials: HTTPAuthorizationCredentials = Depends(_security)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)